scanners/mobilefriendly.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from scanners import utils
+import json
+import os
+import requests
+
+##
+# == pageload ==
+#
+# Evaluate page laod time information using Phantomas.
+#
+# If data exists for a domain from `inspect`, will use the
+# previously detected "canonical" endpoint for a domain.
+##
+
+init = None
+
+workers = 4
+
+def scan(domain, options):
+    logging.debug("[%s][pageload]" % domain)
+
+    inspection = utils.data_for(domain, "inspect")
+    
+    # TODO: Factor out the next 22 lines.
+
+    # If we have data from inspect, skip if it's not a live domain.
+    if inspection and (not inspection.get("up")):
+        logging.debug("\tSkipping, domain not reachable during inspection.")
+        return None
+
+    # If we have data from inspect, skip if it's just a redirector.
+    if inspection and (inspection.get("redirect") is True):
+        logging.debug("\tSkipping, domain seen as just a redirector during inspection.")
+        return None
+
+    # We need a URL, not just a domain.
+    if not (domain.startswith('http://') or domain.startswith('https://')):
+
+        # If we have data from inspect, use the canonical endpoint.
+        if inspection and inspection.get("canonical"):
+            url = inspection.get("canonical")
+
+        # Otherwise, well, whatever.
+        else:
+            url = 'http://' + domain
+    else:
+        url = domain
+
+    # We'll cache prettified JSON from the output.
+    cache = utils.cache_path(domain, "mobilefriendly")
+
+    # If we've got it cached, use that.
+    if (options.get("force", False) is False) and (os.path.exists(cache)):
+        logging.debug("\tCached.")
+        raw = open(cache).read()
+        data = json.loads(raw)
+        if data.get('invalid'):
+            return None
+
+    # If no cache, or we should run anyway, do the scan.
+    else:
+        logging.debug("\t requests.get %s" % (url))
+        params = {
+#           'key': 'AIzaSyDkEX-f1JNLQLC164SZaobALqFv4PHV-kA',  # not needed
```

Comment:
There's a commented out API key here? Is this a valid key? If so, you should revoke it immediately.